### PR TITLE
Adds PWA support, fixes some lighthouse errors

### PR DESCRIPTION
**Pull request by georgiemathews**
App now works offline.
Before: 
image

After:
image

### CODE REVIEW

**Comment by reimertz** [comment]
This breaks the water egg. :(

File: docs/index.html [hunk 1]
```diff
@@ -55,6 +58,10 @@
     </footer>
 </body>
 <script type="text/javascript">
-console.log("%c\t\t\t\t\t\t\t\t\t\t\t\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n👋 Hi there! 👋\n\nIt seems you like to poke around as much as we do.\n\nhttps://github.com/Microsoft/join-dev-design", "color: white; background-color: #080808; background-image: url(https://microsoft.github.io/join-dev-design/msft-square.svg); background-size: 70%; background-position: 50% 25%; background-repeat: no-repeat; padding: 3em; text-align: center;font-size: 1.25em;");
+    if ('serviceWorker' in navigator) {
+        navigator.serviceWorker.register('/service-worker.js');
+    }
+    // console.log("%c\t\t\t\t\t\t\t\t\t\t\t\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n👋 Hi there! 👋\n\nIt seems you like to poke around as much as we do.\n\nhttps://github.com/Microsoft/join-dev-design", "color: white; background-color: #080808; background-image: url(https://microsoft.github.io/join-dev-design/msft-square.svg); background-size: 70%; background-position: 50% 25%; background-repeat: no-repeat; padding: 3em; text-align: center;font-size: 1.25em;");
```
Comment (reimertz) [inline docs/index.html]: I assume this was removed so you could monitor console output but please enable again :)

**Comment by georgiemathews** [comment]
image
Rebased, and ready to merge! Let me know if any additional changes needed.

**Comment by dvdsgl** [comment]
Strange, there are many unrelated changes and some conflicts.

I'm not familiar with the effects of adding PWA support. Does this mean visitors will see a cached version of the site?

**Comment by georgiemathews** [comment]
@dvdsgl one of the main benefits here would be being able to see the site even while offline via, as you mentioned, a cache. Since all of our assets are static, we can easily invalidate the cache through the serviceworker each time there is a new deploy.
You can read more here: https://developers.google.com/web/progressive-web-apps/

**Comment by dvdsgl** [comment]
@prichodko thoughts?

**Comment by prichodko** [comment]
It could be a _nice_ thing to have, but as a name suggests it makes a progressive **web app**, app which users will be able to add to their phone home screens and/or use offline. We however have a simple static site, which nobody will use in a way PWA's are designed. So IMHO it's an overkill, but on the other side why not 😄.

Note: This PR also deletes an `.editorconfig` and breaks formatting in every file. I suppose it wasn't intentional, but it definitely blocks PR from merging.

**Comment by dvdsgl** [comment]
Yes, I'd like to leave out PWA support for now.

@georgiemathews thank you for your work!

**Comment by Rob-Rychs** [comment]
I like where your head was at @georgiemathews 👍

**Pull request closed without merging**